htdocs/core/lib/files.lib.php

```diff
@@ -152,7 +152,7 @@ function dol_dir_list($utf8_path, $types = "all", $recursive = 0, $filter = "",
 
 				// Check if file is qualified
 				foreach ($excludefilterarray as $filt) {
-					if (preg_match('/'.$filt.'/i', $utf8_file) || preg_match('/'.$filt.'/i', $utf8_fullpathfile)) {
+					if (preg_match('{'.$filt.'}i', $utf8_file) || preg_match('{'.$filt.'}i', $utf8_fullpathfile)) {
```

Comment:
This will break compatibility with current format of the parameter $excludefilter that expect to have string escaped for "/" regex separator. It is commonly used this way by a lot of external modules (and may be dolibarr itself), so i prefer to cancel PR. We have now a protection that may be enough.